core/src/err.rs

```diff
@@ -1,15 +1,27 @@
-use constants::{GENERAL_ERROR, TIMEOUT_EXPIRED, UNABLE_TO_CONNECT};
+use constants::{
+    COLUMN_NOT_FOUND, GENERAL_ERROR, NO_DSN_OR_DRIVER, TIMEOUT_EXPIRED, UNABLE_TO_CONNECT,
+};
 use mongodb::error::{BulkWriteFailure, ErrorKind, WriteFailure};
 use thiserror::Error;
 
 pub type Result<T> = std::result::Result<T, Error>;
 
 #[derive(Error, Debug)]
 pub enum Error {
+    #[error(transparent)]
+    BsonDeserialization(#[from] bson::de::Error),
```

Comment:
Personal preference is to omit the "Error" suffix from the variant names of the `Error` enum since the suffix is redundant. This is inconsistent with the two existing `Error` variants: `MongoParseConnectionStringError` and `MongoError`. I can make these all consistent if anyone prefers and can do that by adding the suffix to the new errors or removing from the old -- just lmk preferences!

Reply:
Omitting the `Error` suffix sounds good as it is redundant here.  This would also be consistent with `ODBCError` in `odbc/src/api/errors.rs` to not append `Error`.